plugins/poll/assets/javascripts/controllers/poll-breakdown.js.es6

```diff
@@ -0,0 +1,83 @@
+import I18n from "I18n";
+import Controller from "@ember/controller";
+import { action } from "@ember/object";
+import { classify } from "@ember/string";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import discourseComputed from "discourse-common/utils/decorators";
+import loadScript from "discourse/lib/load-script";
+
+export default Controller.extend(ModalFunctionality, {
+  model: null,
+  charts: null,
+  groupedBy: null,
+  highlightedOption: null,
+  displayMode: "percentage",
+
+  @discourseComputed("model.groupableUserFields")
+  groupableUserFields(fields) {
+    return fields.map(field => {
+      const transformed = field.split("_").filter(Boolean);
+
+      if (transformed.length > 1) {
+        transformed[0] = classify(transformed[0]);
+      }
+
+      return { id: field, label: transformed.join(" ") };
+    });
+  },
+
+  @discourseComputed("model.poll.options")
+  totalVotes(options) {
+    return options.reduce((sum, option) => sum + option.votes, 0);
+  },
+
+  onShow() {
+    this.set("charts", null);
```

Comment:
setProperties ?

Reply:
I'm actively avoiding `setProperties` to make it easier for future-me to update all `this.set(x, y)` calls with Octane's `this.x = y` 😁

Reply:
OK makes sense :+1: